### PR TITLE
SOL-150707: Updating get-broker-health to be get-broker-status

### PR DESCRIPTION
**Pull request by solace-awaheed**
## Summary
Rename the `get-broker-health` MCP tool to `get-broker-status`, removing the "health" claim. The tools will return a useful snapshot of broker operational status. 

Fixes # SOL-150707

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Dependency update

## Motivation and Context

Claiming we know full broker health was an over claim, brought it down a notch to say status instead. The next step as part of this command will be to add a hardware level debugging tool for appliances. 

Note: The description update doesn't mention the get-hardware-details yet, that will be added as part of a follow up story once the tool is implemented. 

## Changes Made

- Package Rename 
- Tool Rename 
- Docs Rename 

## Testing

- Unit testing 
- Made sure we can still build 
- Tested out locally to make sure we call "get-broker-status" instead of "get-broker-health" 

**Test Environment:**
- Go version:
- OS:
- Broker version (if applicable):

**Tests Performed:**
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated (if applicable)
- [ ] E2E tests added/updated (if applicable)
- [ ] Tested locally with `go test -race ./...`
- [ ] Tested with real Solace broker

**Test Coverage:**

-

## Breaking Changes

**Impact:**

**Migration Guide:**

### Code Quality
- [ ] Code follows the [coding standards](../.github/CONTRIBUTING.md#coding-standards)
- [ ] Self-review completed (checked for typos, logic errors, edge cases)
- [ ] Code is well-commented (explains "why", not "what")
- [ ] Complex logic has explanatory comments
- [ ] No commented-out code or debug statements left in

### Security
- [ ] **No credentials in code** (checked all files, commits, and logs)
- [ ] Followed [secure logging rules](../docs/secure-logging-rules.md)
- [ ] Credential-carrying types implement `slog.LogValuer`
- [ ] No security vulnerabilities introduced (checked with golangci-lint/gosec)

### Testing
- [ ] All tests pass locally: `go test -race ./...`
- [ ] No race conditions detected
- [ ] Edge cases covered by tests
- [ ] Error paths tested
- [ ] Test names clearly describe what is being tested

### Documentation
- [ ] README.md updated (if user-facing changes)
- [ ] docs/ updated (if architecture/design changes)
- [ ] godoc comments added/updated for exported functions
- [ ] CHANGELOG.md updated (if applicable)
- [ ] Examples updated (if applicable)

### Git & CI
- [ ] All commits are signed off (DCO: `git commit -s`)
- [ ] Commits have clear, descriptive messages
- [ ] Branch is up to date with main (rebased if needed)
- [ ] No merge conflicts
- [ ] CI checks passing (lint, build, test, E2E)

### Breaking Changes (if applicable)
- [ ] Breaking changes documented in commit message
- [ ] Breaking changes documented in PR description
- [ ] Migration guide provided
- [ ] Deprecated functionality marked with comments and deprecation notices

## Related Issues/PRs

- Related to #
- Depends on #
- Blocks #

---

**For Reviewers:**

**Focus Areas**: 

**Questions**:

### CODE REVIEW

**Review by solace-aross — DISMISSED** [review]
(Withdrawn. Superseded by a fresh review below.)

File: internal/tools/sempv1/brokerstatus/handler.go [hunk 1]
```diff
@@ -58,15 +58,16 @@ func NewHandler() *Handler {
...
+		Description: "Returns a curated point-in-time status snapshot of a Solace " +
+			"broker's operational state - edition and version, uptime and restart " +
+			"reason, scaling limits and resource headroom (to flag under-scaling), " +
+			"memory and message-spool utilization. Reports raw state, not a " +
+			"pass/fail verdict - interpretation depends on deployment intent " +
+			"(e.g. message-spool disabled is expected for direct-messaging-only " +
+			"deployments, not a fault). Use whenever the user asks about a " +
+			"broker's status, whether it is slow, restarted, under-scaled, low " +
+			"on capacity, or before maintenance. Specify the target broker by " +
+			"its configured alias.",
```
Comment (solace-aross) [inline internal/tools/sempv1/brokerstatus/handler.go]: Agreed on the framing elsewhere in this description: no implied health verdict. The risk in dropping the word "health" entirely is the opposite failure. "Is my broker healthy?" is still what operators will type, and after this merge no broker-level tool description matches it (`get-vpn-health` is the closest lexical match, wrong scope). The LLM then either picks the wrong tool or answers with no data, improvising exactly the health judgment this PR is trying not to make. Routing the question to real status data is what keeps the model honest.

The keyword and the claim are separable: the description already says "Reports raw state, not a pass/fail verdict." Suggest the minimal edit to the routing sentence: "Use whenever the user asks about a broker's status or health, whether it is slow, restarted, under-scaled, low on capacity, or before maintenance." That references the user's question, not a capability we don't have.

**Review by solace-aross — CHANGES_REQUESTED** [review]
The rename itself is clean. I verified the branch builds and `go test -race ./...` passes locally, the e2e scripts don't assert this tool name, and no stray `brokerhealth` references remain outside historical planning docs.

Three things:

1. The inline comment on the tool description is the one I'd hold the merge on. It's a tool-routing regression, not wording. Short version: keep the "no verdict" framing, but keep "health" as a routing keyword so the question lands on real data.

2. Breaking Changes: since the server is unreleased, suggest one line in the Impact field, e.g. "Pre-release; no external consumers, no migration needed." Otherwise the checked box reads as an open question later. Non-blocking.

3. Copilot's six collapsed "low confidence" comments are right, and you're touching those lines anyway: the `docs/semp/...` references should be `docs/internal/semp/...`, and `spool_response.go:23` still says "is the broker healthy?". One more in the same family: `internal/composite/definitions/tools.yaml:10` still coaches future tool authors to write "assess overall broker health" in descriptions, which contradicts the stance this PR takes.

**Comment by bczoma** [comment]
I was expecting a tool-name related change in one of the e2e test folders but at the same time e2e tests are passing.
@helen-zhu-solace is kindly taking a look and will advise if we should add coverage.

**Comment by helen-zhu-solace** [comment]
get-broker-status will be added to e2e-monitoring after this PR is merged [SOL-150724](https://sol-jira.atlassian.net/browse/SOL-150724)

[SOL-150724]: https://sol-jira.atlassian.net/browse/SOL-150724?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Review by solace-aross — APPROVED** [review]
All review points are addressed. The tool description now keeps "health" as a routing keyword while disclaiming a verdict, which resolves the blocking comment. The spool_response.go wording, the tools.yaml guidance, and the doc paths are all fixed. I rebuilt the branch and ran go test -race ./... locally; everything passes, and the new 17-tool count matches what the server actually registers.

Two small leftovers, neither blocking:
- README line 67: the architecture diagram still says "13 read-only tools" while the other two spots now say 17. Fine to fix here or in a follow-up.
- The Breaking Changes Impact field in the description is still empty; a one-liner like "Pre-release; no external consumers" would close it out.
